op-node/node/node.go

```diff
@@ -483,30 +487,35 @@ func (n *OpNode) initPProf(cfg *Config) error {
 	return nil
 }
 
-func (n *OpNode) initP2P(ctx context.Context, cfg *Config) error {
-	if cfg.P2P != nil {
+func (n *OpNode) p2pEnabled() bool {
+	return n.cfg.P2P != nil
```

Comment:
It may be nicer to add this method directly to the `Config` struct as a public method `P2PEnabled() bool`. Then you could use it in setup code instead and don't need to retain the `cfg` in the `OpNode` struct.

Reply:
I think you're right, thanks.

Reply:
I see why I did it, it's because the same reason I wanted to test the config for features rather than the node state might apply to other features later.